lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/TableExceptionTranslatorImpl.java

```diff
@@ -165,4 +185,21 @@ private static String appendUnquotedKeyWordMessage(String input) {
 		}
 		return input;
 	}
+	
+	/**
+	 * Extract the constraint name from a raw error message
+	 * @param rawMessage
+	 * @return
+	 */
+	public static Optional<String> getConstraintViolationName(String rawMessage){
+		int start = rawMessage.indexOf(CHECK_CONSTRAINT_PREFIX);
+		if(start > -1) {
+			start += CHECK_CONSTRAINT_PREFIX.length();
+			int end = rawMessage.indexOf(CHECK_CONSTRAINT_SUFFIX);
+			if(end > 0) {
+				return Optional.of(rawMessage.substring(start, end));

```

Comment:
Regex such as: "Check constraint '(.+)' is violated.". We could check if we could make use of a MySQL error code.